Fix removal of old team members when new are added

dlukman opened the pull request:
The grafana_team resource removes existing users from teams when new ones are added. This change should fix it.

How to reproduce the issue:
- Add new members to existing team members using the grafana_team resource and apply
- Run terraform plan and observe that all users previously there were removed and want to be re-added
- Run again, the old ones are re-added, the new ones are removed (repeats on each rerun)

TF version: 1.3.7
Grafana version: 9.1.5

dlukman-soundhound committed:
Fix removal of old team members when new are added

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/terraform-provider-grafana?pullRequest=795) <br/>All committers have signed the CLA.

julienduchesne commented:
Did you test this fix? It shouldn't do anything, IIUC, because new == old is implicit, it's kind of a duplicate condition right now

dlukman commented:
Yes, I tested it. The removal of this condition made it behave as expected.

JuanJoanJon commented:
We need this ASAP :-)

julienduchesne commented:
> We need this ASAP :-)

I'd like to have a test on this before merging. From the description of it, it should be testable

pascal-hofmann commented:
[This](https://github.com/grafana/terraform-provider-grafana/pull/780/) introduced the issue and did not need tests. It would be great to have a consistent policy for PRs. 😬

Maybe just revert #780 for now?

Edit: I just checked. The issue is not present in version 1.33.0 of the provider.

dlukman commented:
@julienduchesne Unfortunately I don't have the time to follow the proper process. I understand it's not optimal. I hoped it would help resolve it faster. If you wish to add tests or improve upon the change, feel free to do so.

I currently have it locked to 1.33.0 which doesn't have the issue as mentioned earlier.

julienduchesne commented:
> @julienduchesne Unfortunately I don't have the time to follow the proper process. I understand it's not optimal. I hoped it would help resolve it faster. If you wish to add tests or improve upon the change, feel free to do so.
> 
> I currently have it locked to 1.33.0 which doesn't have the issue as mentioned earlier.

I got it. I'll try to reproduce this in tests

julienduchesne committed:
Add test case about adding a team member
Also, reformat the entire team test suite:
- Allow parallel testing (unique team and user names)
- Merge all test definitions into a function

julienduchesne commented:
I added a test that reproduces the bug. Confirmed that the fix works! Thanks @dlukman
